### PR TITLE
CNN support

### PR DESCRIPTION
**Pull request by souravsanyal06**
updated source code to run cnn along with instructions in test/conv

### CODE REVIEW

File: how_to_run.md [hunk 1]
```diff
@@ -52,9 +52,17 @@ make
 cd ../test/
 export LD_LIBRARY_PATH=`pwd`/../src:$LD_LIBRARY_PATH
 ```
+#### 4.3 - Compile the examples:
+### To run with weights, train a model in any deep-learning framework (like py-torch or tensorflow) to get the weights.
+### A "name of model".cpp file and a weights folder is required in the Puma Compiler /test/.
+### A sample mlp_l4_mnist.cpp and mlp_l4_mnist_weights/ is provided in Puma Simulator/test/mlp_l4_mnist/ as a template which needs to be copied to Puma Compiler/test or
+### Create your model.cpp (which will have same configuration as the one used to get weights in pytorch)  and weights folder (obtained from pytorch) in Puma Compiler/test
+```
```
Comment (Aayush-Ankit) [inline how_to_run.md:60]: Why is the Section 4.3 (Compile with examples) here before Section 4.2 ?
Is the goal here to re-organize the steps for "run with weights" for readability ?

File: include/config.py [hunk 2]
```diff
@@ -102,7 +102,11 @@
 # (b bit of address = logN, N is the number of nodes)
 
 # Change here - Specify the Node parameters here
-num_tile_compute = 7 # number of tiles mapped by dnn (leaving input and output tiles)
+#<<<<<<< HEAD
+num_tile_compute =4 # number of tiles mapped by dnn (leaving input and output tiles)
+#=======
+#num_tile_compute = 4 # number of tiles mapped by dnn (leaving input and output tiles)
+#>>>>>>> f4f8f41269a93523c87f38d8325427df763a4d4a
```
Comment (Aayush-Ankit) [inline include/config.py]: Changes in config file parameters shouldn't me merged.

File: include/config.py [hunk 3]
```diff
@@ -4,7 +4,7 @@
 ## Debug - 0 (1): dpe simulation will (won't) produce ima/tile traces while simulating
 cycles_max = 5000000 # Put both these to very large numbers (when design is bug-free)!
 debug = 1
-xbar_record = 1
+xbar_record = 0
```
Comment (Aayush-Ankit) [inline include/config.py]: Changes in config file parameters shouldn't me merged.

File: src/dpe.py [hunk 4]
```diff
@@ -192,8 +192,9 @@ def run(self, net):
     parser = argparse.ArgumentParser()
     parser.add_argument(
         "-n", "--net", help="The net name as it is in test/testasm.", default='large')
-    parser.add_argument(
-            "-t", "--tile", help ="The number of tiles as generated by compiler.", default=-1)
```
Comment (Aayush-Ankit) [inline src/dpe.py]: I understand this feature doesn't work at the moment. Still it should not be removed - a previous pull request resolved this issue, see Lines 218-221 below.

File: src/dpe.py [hunk 5]
```diff
@@ -215,17 +216,15 @@ def run(self, net):
 
     total_tiles = count_tiles(model_path) - 2
     print(total_tiles)
-
-    if(args.tile != -1):
-        total_tiles = int(args.tile)
-
```
Comment (Aayush-Ankit) [inline src/dpe.py]: Don't remove the past additions (probably get the most recent master), before merging your local commit, and then followed by pull request.

File: src/dpe.py [hunk 6]
```diff
@@ -192,8 +192,9 @@ def run(self, net):
     parser = argparse.ArgumentParser()
     parser.add_argument(
         "-n", "--net", help="The net name as it is in test/testasm.", default='large')
-    parser.add_argument(
-            "-t", "--tile", help ="The number of tiles as generated by compiler.", default=-1)
+    args = parser.parse_args()
+    net = args.net
+
```
Comment (Aayush-Ankit) [inline src/dpe.py:197]: Remove

File: src/dpe.py [hunk 7]
```diff
@@ -215,17 +216,15 @@ def run(self, net):
 
     total_tiles = count_tiles(model_path) - 2
     print(total_tiles)
-
-    if(args.tile != -1):
-        total_tiles = int(args.tile)
-
+
```
Comment (Aayush-Ankit) [inline src/dpe.py:219]: Remove

File: test/conv/how-to-run-cnn.md [hunk 8]
```diff
@@ -0,0 +1,30 @@
+1. Copy the sample cnnbig.cpp and paste it in puma-compiler/test/
```
Comment (Aayush-Ankit) [inline test/conv/how-to-run-cnn.md:1]: Shouldn't this (extra steps/caution in CNN) come as a separate section in "how_to_run.md" itself ?
Lines 13-end are repeated.

File: test/conv/how-to-run-cnn.md [hunk 9]
```diff
@@ -0,0 +1,30 @@
...
+		make 'net'.test
+		./'net'.test
+		rm -rf 'net'
+		./generate-py.sh
+
+4. Copy the 'net' directory to puma-simulator/test/testasm/
+
+5. Copy the input.npy in puma-simulator/test/utils/ and paste it in puma-simulator/test/testasm/'net'/
+
+
```
Comment (Aayush-Ankit) [inline test/conv/how-to-run-cnn.md]: Shouldn't this (extra steps/caution in CNN) come as a separate section in "how_to_run.md" itself ?
Lines 13-end are repeated.

File: how_to_run.md [hunk 10]
```diff
@@ -52,9 +52,17 @@ make
...
 
 #### 4.2 - Compile model:
 
+*Note: To run cnns, consult the test/conv directory
```
Comment (Aayush-Ankit) [inline how_to_run.md]: Shouldn't this be pointing to steps in "how-to-run-cnn.md" ?

**Comment by souravsanyal06** [comment]
squashed

**Pull request closed without merging**